### ISSUE TITLE
False positive on 'abstract-method' when extending ABC classes

### ISSUE THREAD

**Issue by jnsnow** [PyCQA/pylint#3098]
### Steps to reproduce

Consider this snippet:

```python
import abc
from collections.abc import Mapping
import logging

class SomeDictMixin(Mapping, abc.ABC):
    def logged_get(self, key, default=None):
        tmp = self.get(key, default)
        logging.debug("Got %s for key %s", tmp, key)
        return tmp
```

The stated purpose of this class would be to create an abstract extension of `Mapping` that provides additional functionality as a "mixin". By deriving from `Mapping` we declare our requirement that `self.get` will be present. By inheriting `abc.ABC` we stipulate that this class itself cannot be instantiated until all of its abstracted methods are defined.

### Current behavior

Pylint does not appear to understand the semantic consequences of inheriting from `abc.ABC`:
```
example.py:5:0: W0223: Method '__getitem__' is abstract in class 'Mapping' but is not overridden (abstract-method)
example.py:5:0: W0223: Method '__iter__' is abstract in class 'Iterable' but is not overridden (abstract-method)
example.py:5:0: W0223: Method '__len__' is abstract in class 'Sized' but is not overridden (abstract-method)
```

Namely, that we are explicitly stating that this class is itself abstract.

### Expected behavior

Pylint will understand that we are implementing an abstract class and not produce these errors.

### pylint --version output

```
pylint 2.3.1
astroid 2.2.5
Python 3.7.4 (default, Jul  9 2019, 16:32:37) 
[GCC 9.1.1 20190503 (Red Hat 9.1.1-1)]
```

**Comment by PCManticore** [comment]
Thanks for opening an issue! 

Right now `pylint` assumes that any subclass of an abstract class needs to implement the corresponding abstract methods, as such `pylint` does not properly support *intermediary* abstract classes, such as this one. It's a long standing issue with how `pylint` assumes that an abstract class should operate, but in this particular case, we might be able to retrieve a hint of its abstractness from the mixin naming.

**Comment by jnsnow** [comment]
It may as well be a core python issue, too. Even if you don't inherit ABC, you can't instantiate the mixin, so there's not really a clean way to semantically declare your intent.

Inheriting ABC does quiet Pycharm's warnings, though, hence that trick. Pycharm seems to use that to understand that the class is intended to be abstract.

That does add useless depth to the MRO, though.

(Sorry for filing so many issues rapid-fire! Thank you for your triage work.)
